src/Vault.sol

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: BSD Clause-3
+pragma solidity ^0.8.24;
+
+import {
+    AccessControlUpgradeable,
+    Address,
+    ERC20PermitUpgradeable,
+    IERC20,
+    Math,
+    ReentrancyGuardUpgradeable,
+    SafeERC20
+} from "./Common.sol";
+
+import {Storage} from "src/Storage.sol";
+import {IVault} from "src/interface/IVault.sol";
+
+contract Vault is IVault, ERC20PermitUpgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
what's the intention on how to handle withdrawals for the Vault?

Reply:
async: A simple withdraw queue would likely be the best phase 1 implementation
sync: Most projects use a buffer of underlying (WETH) that sits in the vault

Reply:
I've updates the vault docs here in this repo to explain how withdraws function. Closing this for now